perun-base/src/main/java/cz/metacentrum/perun/core/api/AttributeDefinition.java

```diff
@@ -176,39 +170,32 @@ public int hashCode() {
 
 	@Override
 	public boolean equals(Object obj) {
-		if(obj == null) return false;
+		if (obj == null) return false;
 
 		if (!(obj instanceof AttributeDefinition)) return false;
 
 		final AttributeDefinition other = (AttributeDefinition) obj;
 
-		if (this.getId() != other.getId()) return false;
-		if(this.friendlyName == null ? other.friendlyName != null : !this.friendlyName.equals(other.friendlyName)) return false;
-
-		return true;
+		return this.getId() == other.getId() && (this.friendlyName == null ? other.friendlyName == null : this.friendlyName.equals(other.friendlyName));
 	}
 
 	@Override
 	public String serializeToString() {
-		StringBuilder str = new StringBuilder();
-
-		return str.append(this.getClass().getSimpleName()).append(":[").append(
-				"id=<").append(getId()).append(">").append(
-				", friendlyName=<").append(getFriendlyName() == null ? "\\0" : BeansUtils.createEscaping(getFriendlyName())).append(">").append(
-				", namespace=<").append(getNamespace() == null ? "\\0" : BeansUtils.createEscaping(getNamespace())).append(">").append(
-				", type=<").append(getType() == null ? "\\0" : BeansUtils.createEscaping(getType())).append(">").append(
-				']').toString();
+		return this.getClass().getSimpleName() + ":[" +
```

Comment:
Yes, actually the compiler converts the concatenation of strings into calling a StringBuilder, so during runtime both versions are identical, and this one is simpler to read in source code. The only place where using StringBuilder explicitly makes sense is when the string is created in a loop.

This change was suggested by Intellij IDEA and its description said that it is suggested only if it is at least as efficient as the original code.